internal/runbits/requirements/requirements.go

```diff
@@ -58,7 +58,7 @@ type configurable interface {
 
 const latestVersion = "latest"
 
-func ExecuteRequirementOperation(prime primeable, requirementName, requirementVersion string, operation model.Operation, nsType model.NamespaceType) (rerr error) {
+func ExecuteRequirementOperation(prime primeable, requirementName, requirementVersion string, bitWidth int, operation model.Operation, nsType model.NamespaceType) (rerr error) {
```

Comment:
Adding the bitWidth here seems to move even more responsibility to this humongous function. And it's an arg that's ONLY relevant to platform requirements. Why not calculate the platform ID before calling this function and call it with the resolved ID?

ie. move this logic to your runner:

https://github.com/ActiveState/cli/pull/2288/files/a074423c8e6b85288c5b0754a88370094465f2f0..ac973fcdd57b19b39bb2ed7b8742627ace0fbb52#diff-d9612958a129acac4c7a4c03e241b6b912c286635b188a24975afe794fcc94f5R750

Then just run `ExecuteRequirementOperation` with the platform ID.

Reply:
Resolving the platform ID in the runner will result in output like `Installing requirement <UUID>`. I would prefer not to have another argument in this function however in this case it is only passed on to lower-level functions. Resolving the ID early could lead to more complexity in order to get the UX right.